cli/pkg/helpers/clients.go

```diff
@@ -0,0 +1,173 @@
+	}
+	return client
+}
+
+func SchemaClient() (v1.SchemaClient, error) {
+	if memoryResourceClient != nil {
+		return v1.NewSchemaClient(memoryResourceClient)
+	}
+
+	cfg, err := kubeutils.GetConfig("", "")
+	if err != nil {
+		return nil, errors.Wrapf(err, "getting kube config")
+	}
+	cache := kube.NewKubeCache(context.TODO())
+	schemaClient, err := v1.NewSchemaClient(&factory.KubeResourceClientFactory{
+		Crd:         v1.SchemaCrd,
+		Cfg:         cfg,
+		SharedCache: cache,
+	})
+	if err != nil {
+		return nil, errors.Wrapf(err, "creating resolver map client")
+	}
+	if err := schemaClient.Register(); err != nil {
+		return nil, err
+	}
+	return schemaClient, nil
+}
+
+func MustResolverMapClient() v1.ResolverMapClient {
+	client, err := ResolverMapClient()
+	if err != nil {
+		log.Fatalf("failed to create resolver map client: %v", err)
+	}
+	return client
+}
+
+func ResolverMapClient() (v1.ResolverMapClient, error) {
+	if memoryResourceClient != nil {
+		return v1.NewResolverMapClient(memoryResourceClient)
+	}
+
+	cfg, err := kubeutils.GetConfig("", "")
+	if err != nil {
+		return nil, errors.Wrapf(err, "getting kube config")
+	}
+	cache := kube.NewKubeCache(context.TODO())
+	resolverMapClient, err := v1.NewResolverMapClient(&factory.KubeResourceClientFactory{
+		Crd:         v1.ResolverMapCrd,
+		Cfg:         cfg,
+		SharedCache: cache,
+	})
+	if err != nil {
+		return nil, errors.Wrapf(err, "creating resolver map client")
+	}
+	if err := resolverMapClient.Register(); err != nil {
+		return nil, err
+	}
+	return resolverMapClient, nil
+}
+
+func MustSettingsClient() glooV1.SettingsClient {
+	client, err := SettingsClient()
+	if err != nil {
+		log.Fatalf("failed to create settings client: %v", err)
+	}
+	return client
+}
+
+func SettingsClient() (glooV1.SettingsClient, error) {
+	if memoryResourceClient != nil {
+		return glooV1.NewSettingsClient(memoryResourceClient)
+	}
+
+	cfg, err := kubeutils.GetConfig("", "")
+	if err != nil {
+		return nil, errors.Wrapf(err, "getting kube config")
+	}
+	cache := kube.NewKubeCache(context.TODO())
+	settingsClient, err := glooV1.NewSettingsClient(&factory.KubeResourceClientFactory{
+		Crd:         glooV1.SettingsCrd,
+		Cfg:         cfg,
+		SharedCache: cache,
+	})
+	if err != nil {
+		return nil, errors.Wrapf(err, "creating settings client")
+	}
+	if err := settingsClient.Register(); err != nil {
+		return nil, err
+	}
+	return settingsClient, nil
+}
+
+func MustSecretClient() glooV1.SecretClient {
+	client, err := secretClient()
+	if err != nil {
+		log.Fatalf("failed to create Secret client: %v", err)
+	}
+	return client
+}
+
+func secretClient() (glooV1.SecretClient, error) {
+	if memoryResourceClient != nil {
+		return glooV1.NewSecretClient(memoryResourceClient)
+	}
+
+	clientset, err := getKubernetesClient()
+	if err != nil {
+		return nil, errors.Wrapf(err, "getting kube config")
+	}
+	secretClient, err := glooV1.NewSecretClient(&factory.KubeSecretClientFactory{
```

Comment:
I think this needs KubeCoreCache

Reply:
Maybe, but I also don't use that code so I removed it